### PR TITLE
Support versioned kv store in vault

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**
- Some refactoring in new vault client.

### CODE REVIEW

File: vault/vault.go [hunk 1]
```diff
@@ -2,192 +2,238 @@ package vault
...
+	return v.backendPath + secretID
+}
+
+func (v *vaultSecrets) isKvBackendV2() (bool, error) {
+	mounts, err := v.client.Sys().ListMounts()
+	if err != nil {
+		return false, err
+	}
+
+	for path, mount := range mounts {
```
Comment (adityadani) [inline vault/vault.go]: Why not find out the backend version only once while creating the client and then use it in each API?

File: vault/vault.go [hunk 2]
```diff
@@ -2,192 +2,238 @@ package vault
...
+		v.backendPath)
+}
+
 func (v *vaultSecrets) GetSecret(
-	secretId string,
+	secretID string,
 	keyContext map[string]string,
 ) (map[string]interface{}, error) {
-	secretValue, err := v.client.Logical().Read(getSecretKey(secretId))
+	isKvV2, err := v.isKvBackendV2()
```
Comment (adityadani) [inline vault/vault.go]: I think we should check the backend while creating a client and if we get an error while listing the mounts, we should error out without returning a client.

Comment (piyush-nimbalkar) [reply vault/vault.go]: Yes, makes sense. Will do the change.

**Comment by piyush-nimbalkar** [comment]
Have updated the review as discussed.

**Pull request merged**